user-ops-indexer/user-ops-indexer-logic/src/indexer/base_indexer.rs

```diff
@@ -0,0 +1,538 @@
+use crate::{indexer::settings::IndexerSettings, repository, types::user_op::UserOp};
+use anyhow::{anyhow, bail};
+use ethers::prelude::{
+    abi::{AbiEncode, Error},
+    parse_log,
+    types::{Action, Address, Bytes, Filter, Log, TransactionReceipt},
+    EthEvent, Middleware, Provider, ProviderError, PubsubClient, H256,
+};
+use futures::{
+    stream,
+    stream::{repeat_with, BoxStream},
+    Stream, StreamExt,
+};
+use sea_orm::DatabaseConnection;
+use std::{future, num::NonZeroUsize, sync::Arc, time, time::Duration};
+use tokio::time::sleep;
+use tracing::instrument;
+
+#[derive(Hash, Eq, PartialEq)]
+struct Job {
+    tx_hash: H256,
+    block_hash: H256,
+}
+
+impl From<H256> for Job {
+    fn from(hash: H256) -> Self {
+        Self {
+            tx_hash: hash,
+            block_hash: H256::zero(),
+        }
+    }
+}
+
+impl TryFrom<Log> for Job {
+    type Error = anyhow::Error;
+
+    fn try_from(log: Log) -> Result<Self, Self::Error> {
+        if log.removed == Some(true) {
+            bail!("unexpected pending log")
+        }
+        let tx_hash = log
+            .transaction_hash
+            .ok_or(anyhow::anyhow!("unexpected pending log"))?;
+        let block_hash = log
+            .block_hash
+            .ok_or(anyhow::anyhow!("unexpected pending log"))?;
+        Ok(Self {
+            tx_hash,
+            block_hash,
+        })
+    }
+}
+
+pub trait IndexerLogic {
```

Comment:
I would rename IndexerLogic to BaseIndexer or rename this file from `base_indexer.rs` to `indexer_logic.rs`

Reply:
oh, we gave IndexerLogic and Indexer, nvm then